src/test/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorderTest.java

```diff
@@ -0,0 +1,48 @@
+package org.jenkinsci.plugins.awsdevicefarm;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.util.VersionNumber;
+import jenkins.model.Jenkins;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.For;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+@For(AWSDeviceFarmRecorder.class)
+public class AWSDeviceFarmRecorderTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    @Issue("JENKINS-50483")
+    public void dataSerializationSmokeTest() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        AWSDeviceFarmRecorder rec = new AWSDeviceFarmRecorder(
+                "TestProjectName", "TestDevicePool",
+                null, null, "APPIUM_JAVA_JUNIT", false, false, null,
+                null, null, null, null, null, null,
+                null, null, null, null, null, null,
```

Comment:
It looks better in IDE with parameter name visualization...

Reply:
OMG, needs `@DataBoundSetter` like I have never seen!